spot-client/src/common/backend/SpotBackendService.js

```diff
@@ -217,13 +224,30 @@ export class SpotBackendService extends Emitter {
         }
 
         if (!registerDevicePromise) {
+            const storedEndpointId = persistence.get(this.endpointIdPersistenceKey);
+
             logger.log('No stored registration', {
+                endpointIdPersistenceKey: this.endpointIdPersistenceKey,
                 pairingCode,
+                storedEndpointId,
                 storedPairingCode: storedRegistration && storedRegistration.pairingCode
             });
+
             registerDevicePromise
-                = registerDevice(`${this.pairingServiceUrl}`, pairingCode)
-                    .catch(error => this._maybeClearRegistration(error));
+                = registerDevice(`${this.pairingServiceUrl}`, pairingCode, storedEndpointId)
+                    .then(registration => {
+                        const { endpointId, ...rest } = registration;
+
+                        if (storedEndpointId !== endpointId) {
+                            logger.log('Storing new endpoint ID', {
+                                endpointId,
+                                endpointIdPersistenceKey: this.endpointIdPersistenceKey
+                            });
+                            persistence.set(this.endpointIdPersistenceKey, endpointId);
```

Comment:
what if we set and log each time no matter what? Could that be safer? Worse?

Reply:
If register request is going to happen, then another message will be logged above which should contain the endpoint ID already(if it exists).
I'm not sure if we should store every time just for safety. Eventually it's been retrieved from storage already, so it it there?